### PR TITLE
fix(list-item): use pointer cursor when selection mode is none

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #6123

## Summary
Add pointer cursor on hover

### CODE REVIEW

File: src/components/list-item/list-item.scss [hunk 1]
```diff
@@ -19,7 +19,7 @@
   );
 }
 .container:hover {
-  @apply bg-foreground-2;
+  @apply bg-foreground-2 cursor-pointer;
```
Comment (driskull) [inline src/components/list-item/list-item.scss:22]: 👍 LOOKS GOOD.

I think we could probably remove this though...

https://github.com/Esri/calcite-components/blob/1fc4e760b8924ef8903f1283b7fa45e6d40e7231/src/components/list-item/list-item.scss#L51-L53

**Review by driskull — APPROVED** [review]
I'm ok with this, my only reservation is that if we allow these list items to basically become buttons when selection is disabled, then I think we lose a11y support because a keyboard user won't know these are actionable.

cc @geospatialem

**Comment by geospatialem** [comment]
@benelan This looks great, nice work! 🎆 

> I'm ok with this, my only reservation is that if we allow these list items to basically become buttons when selection is disabled, then I think we lose a11y support because a keyboard user won't know these are actionable.

Agree with Matt's comment regarding disabled states and a11y if we try to implement in the future.

**Comment by geospatialem** [comment]
@benelan Can this be merged to verify 6123?

**Comment by benelan** [comment]
Woops sorry I thought I already merged it lol

**Pull request merged**